src/sentry/remote_subscriptions/consumers/result_consumer.py

```diff
@@ -0,0 +1,96 @@
+from __future__ import annotations
+
+import abc
+import logging
+from collections.abc import Mapping
+from typing import Generic, TypeVar
+
+from arroyo.backends.kafka.consumer import KafkaPayload
+from arroyo.processing.strategies.abstract import ProcessingStrategy, ProcessingStrategyFactory
+from arroyo.processing.strategies.commit import CommitOffsets
+from arroyo.processing.strategies.run_task import RunTask
+from arroyo.types import BrokerValue, Commit, FilteredPayload, Message, Partition
+from sentry_kafka_schemas.codecs import Codec
+
+from sentry.conf.types.kafka_definition import Topic, get_topic_codec
+from sentry.remote_subscriptions.models import BaseRemoteSubscription
+
+logger = logging.getLogger(__name__)
+
+T = TypeVar("T")
+U = TypeVar("U", bound=BaseRemoteSubscription)
+
+FAKE_SUBSCRIPTION_ID = 12345
+
+
+class ResultProcessor(abc.ABC, Generic[T, U]):
+    def __init__(self, codec: Codec[T]):
+        self.codec = codec
+
+    @property
+    @abc.abstractmethod
+    def subscription_model(self) -> type[U]:
+        pass
+
+    def __call__(self, message: Message[KafkaPayload | FilteredPayload]):
+        assert not isinstance(message.payload, FilteredPayload)
+        assert isinstance(message.value, BrokerValue)
+
+        try:
+            result = self.codec.decode(message.payload.value)
+        except Exception:
+            logger.exception(
+                "Failed to decode message payload",
+                extra={"payload": message.payload.value},
+            )
+        try:
+            # TODO: Handle subscription not existing - we should remove the subscription from
+            # the remote system in that case.
+            self.handle_result(self.get_subscription(result), result)
+        except Exception:
+            logger.exception("Failed to process message result")
+
+    def get_subscription(self, result: T) -> U:
+        try:
+            subscription = self.subscription_model.objects.get_from_cache(
+                subscription_id=self.get_subscription_id(result)
+            )
+        except self.subscription_model.DoesNotExist:
+            # XXX: Create fake rows for now
+            subscription = self.subscription_model(
+                id=FAKE_SUBSCRIPTION_ID,
+                subscription_id=self.get_subscription_id(result),
+                type="test",
+                url="https://sentry.io/",
+                interval_seconds=300,
+                timeout_ms=500,
+            )
+        return subscription
+
+    @abc.abstractmethod
+    def get_subscription_id(self, result: T) -> str:
+        pass
+
+    @abc.abstractmethod
+    def handle_result(self, subscription: U, result: T):
+        pass
+
+
+class ResultsStrategyFactory(ProcessingStrategyFactory[KafkaPayload], Generic[T, U]):
+    def __init__(self, topic: Topic) -> None:
```

Comment:
Where does this get passed from?

Reply:
Oops, yeah this wasn tested and doesn't work. Adding a test that covers this and fixing